Reapply "[msan] Add avx512-intrinsics.ll and avx512-intrinsics-upgrade.ll test case (#123980)"

thurstond opened the pull request:
This reverts commit b2647ffbf797dd5a457b6b19faab06956934d067 i.e., relands https://github.com/llvm/llvm-project/commit/980e86f130eea02bd41b887f4ed896340fc90f6c.

I had reverted the original patch because of buildbot failures (e.g., https://lab.llvm.org/buildbot/#/builders/13/builds/4938/steps/5/logs/FAIL__LLVM__avx512-intrinsics-upgrade_ll). This reland addresses the likely root cause; namely, that the target datalayout had not been specified.

thurstond committed:
[msan] Add avx512-intrinsics-upgrade.ll test case

This is forked from llvm/test/CodeGen/X86/avx512-intrinsics-upgrade.ll.
It tests intrinsics that LLVM "auto-upgrades"; for example,
@llvm.x86.avx512.mask.store is converted into @llvm.masked.store (which has the interesting side effect that MemorySanitizer can already handle it via its existing handleMaskedStore).

thurstond committed:
Add avx512-intrinsics.ll

thurstond committed:
Move to X86 subdirectory

thurstond committed:
Fix source information

vitalybuka commented:
> I had reverted the original patch because of buildbot failures (e.g., https://lab.llvm.org/buildbot/#/builders/13/builds/4938/steps/5/logs/FAIL__LLVM__avx512-intrinsics-upgrade_ll). This reland addresses the likely root cause; namely, that the target datalayout had not been specified.

It can be more convenient to review reland, it consist of 2 commits, "revert of revert", and fixes
however, if fix is trivial, no need to request a new review

Is possible to fix "undef deprecator" ?

thurstond commented:
> > I had reverted the original patch because of buildbot failures (e.g., https://lab.llvm.org/buildbot/#/builders/13/builds/4938/steps/5/logs/FAIL__LLVM__avx512-intrinsics-upgrade_ll). This reland addresses the likely root cause; namely, that the target datalayout had not been specified.
> 
> It can be more convenient to review reland, it consist of 2 commits, "revert of revert", and fixes however, if fix is trivial, no need to request a new review

I've split the commit into "revert of revert" + the fix.

> Is possible to fix "undef deprecator" ?

Since these are MSan tests, and we want to know how MSan deals with uninitialized memory, undef seems appropriate?

thurstond committed:
Replace undef with extra_param in avx512-intrinsics.ll

thurstond committed:
Replace undef with %extra_param in avx512-intrinsics-upgrade.ll

thurstond committed:
Add extra_param2

thurstond committed:
Add target datalayout

thurstond commented:
> > Since these are MSan tests, and we want to know how MSan deals with uninitialized memory, undef seems appropriate?
> 
> Undef handling is simpler and have dedicated tests. Here we'd like to see how msan propagate shadow, parameter is actually better.

Fixed in https://github.com/llvm/llvm-project/pull/124500/commits/70668afe100362275034b402094b5e5db1939611, https://github.com/llvm/llvm-project/pull/124500/commits/d0fab8404cde23b57fa44fb84f7f5469dfaf60d9 and https://github.com/llvm/llvm-project/pull/124500/commits/3938a45bc0e2c9491713fbe06bf61205c383b1e9

thurstond commented:
Failing tests in CI (https://buildkite.com/llvm-project/github-pull-requests/builds/141497) are unrelated:

LLVM.CodeGen/AArch64/aarch64-build-attributes-all.ll
LLVM.CodeGen/AArch64/aarch64-build-attributes-bti.ll
LLVM.CodeGen/AArch64/aarch64-build-attributes-gcs.ll
LLVM.CodeGen/AArch64/aarch64-build-attributes-pac.ll
LLVM.CodeGen/AArch64/aarch64-build-attributes-pauthabi.ll
